### PR TITLE
(Bug) show the initial and end parts of exec-id at invest page

### LINKED ISSUE #897: (Bug) show the initial and end parts of exec-id at invest page
**Problem**: 
screen shot 2018-05-22 at 16 04 49
Regarding to [web3 design principles](https://medium.com/@lyricalpolymath/web3-design-principles-f21db2f240c1)
>show a compact versions of the hashes but always show the initial and end parts

**Solution**:
implement custom function to show exec-id like this:
`0x06307c94a0785984a070ac...73ce996835c6389da276a4939`. Copy action should copy the whole hash.

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #897 

- Added custom function
- Added test for custom function
- Apply custom function
- Small refactoring in the utils test

Image showing the problem solved:
screenshot_20180628_125416

### CODE REVIEW

**Comment by mariano-aguero** [comment]
@fernandomg 
- Added a button to copy the whole hash.
- Added tooltip to the button
- Added npm script to run sass in a nice way

Image showing the button close to "Crowdsale Execution ID":
screenshot_20180628_181924

Image showing the tooltip:
screenshot_20180628_182021

Regards

File: src/components/contribute/index.js [hunk 1]
```diff
@@ -479,6 +488,17 @@ export class Contribute extends React.Component {
       'qr-selected': contributeThrough === CONTRIBUTION_OPTIONS.QR
     })
 
+    const crowdsaleAddress = (crowdsale && crowdsale.execID) || (MintedCappedProxy && MintedCappedProxy.addr)
+    const crowdsaleAddressTruncated =
+      (crowdsale && truncateStringInTheMiddle(crowdsale.execID)) ||
+      (MintedCappedProxy && truncateStringInTheMiddle(MintedCappedProxy.addr))
```
Comment (vbaranov) [inline src/components/contribute/index.js]: @mariano-aguero let's truncate only for execution id. I mean not truncate in the middle, if it is a proxy address:
```
const crowdsaleAddressTruncated =
      (crowdsale && truncateStringInTheMiddle(crowdsale.execID)) ||
      (MintedCappedProxy && MintedCappedProxy.addr)
```

File: src/components/contribute/index.js [hunk 2]
```diff
@@ -479,6 +488,17 @@ export class Contribute extends React.Component {
...
+    const crowdsaleAddressDescription = crowdsale
+      ? crowdsale.execID
+        ? 'Crowdsale Execution ID'
+        : 'Crowdsale Proxy Address'
+      : 'Crowdsale ID'
+    const crowdsaleAddressTooltip = `Crowdsale execution ID to copy: ${crowdsaleAddress}`
```
Comment (vbaranov) [inline src/components/contribute/index.js]: @mariano-aguero for execution ID the tooltip text should be `Crowdsale execution ID to copy:`. But for proxy address - `Crowdsale proxy address to copy:`

File: src/components/contribute/index.js (hunk 1 above)
Comment (mariano-aguero) [reply src/components/contribute/index.js]: Hi @vbaranov , changes made. Thanks for the feedback

File: src/components/contribute/index.js (hunk 2 above)
Comment (mariano-aguero) [reply src/components/contribute/index.js]: Hi @vbaranov , changes made. Thanks for the feedback

**Comment by vbaranov** [comment]
@dennis00010011b should e2e tests be changed from your side (copy button was added to contribute page)? Or could we safely merge it?

**Comment by dennis00010011b** [comment]
@vbaranov You can merge it, these changes shouldn't affect to e2e

**Pull request merged**